client/src/app/templates/edit/template-edit.component.ts

```diff
@@ -0,0 +1,164 @@
+            this.servicesFormArray,
+        ]);
+        this.generalFormGroup = this.fb.group({
+            generalFormArray: this.generalFormArray,
+        });
+    }
+
+    ngOnInit(): void {
+        this.route.params.subscribe(params => {
+            this.templateId = params['id'];
+            this.templateService.getById(params['id']).subscribe((template: Template) => {
+                this.templateNameFormGroup.controls.templateNameFormControl.setValue(template.name);
+                this.addServiceFormFromTemplate(template);
+                this.sortServiceForm();
+            });
+        });
+
+    }
+
+    addServiceFormFromTemplate(template: Template): void {
+        template.services.forEach(service => {
+            const startTime = moment().hour(service.startAt.hours as number).minute(service.startAt.minutes as number);
+            const endTime = moment().hour(service.endAt.hours as number).minute(service.endAt.minutes as number);
+            const serviceFormGroup = this.fb.group({
+                startFormControl: [startTime.format('HH:mm'), Validators.required],
+                startDayFormControl: [service.startAt.day.toString(), Validators.required],
+                endFormControl: [endTime.format('HH:mm'), Validators.required],
+                endDayFormControl: [service.endAt.day.toString(), Validators.required],
+                nbMaxFormControl: [service.nbMax, Validators.required],
+            });
+            serviceFormGroup.valueChanges.subscribe(() => {
+                this.sortServiceForm();
+            });
+            this.servicesFormArray.push(serviceFormGroup);
+        });
+    }
+
+    addEmptyServiceForm(): void {
+        const serviceFormGroup = this.fb.group({
+            startFormControl: ['', Validators.required],
+            startDayFormControl: ['', Validators.required],
+            endFormControl: ['', Validators.required],
+            endDayFormControl: ['', Validators.required],
+            nbMaxFormControl: ['', Validators.required],
+        });
+        serviceFormGroup.valueChanges.subscribe(() => {
+            this.sortServiceForm();
+        });
+        this.servicesFormArray.push(serviceFormGroup);
+    }
+
+    updateTemplate(): void {
+        const template = new Template();
+        template.id = this.templateId;
+        template.name = this.templateNameFormGroup.controls.templateNameFormControl.value;
+        template.services = this.servicesFormArray.controls.map(formGroup => {
+            return this.prepareService((formGroup as FormGroup).controls);
+        });
+        this.templateService.update(template).subscribe(() => {
+            this.toasterService.showToaster('Template modifié');
+            this.router.navigate(['/templates']);
+        });
+    }
+
+    prepareService(controls): {nbMax: Number,
+        startAt: {day: Number, hours: Number, minutes: Number},
+        endAt: {day: Number, hours: Number, minutes: Number}} {
```

Comment:
It's weird to have the object described here

Reply:
You advise to create a new model for TemplateUnit ?

Reply:
For this one yes we could

Reply:
I've created an interface inside the file.